exporter/awsemfexporter/metric_translator.go

```diff
@@ -159,10 +168,101 @@ func TranslateOtToCWMetric(rm *pdata.ResourceMetrics, dimensionRollupOption stri
 				continue
 			}
 			cwMetrics := getCWMetrics(&metric, namespace, instrumentationLibName, dimensionRollupOption)
-			cwMetricList = append(cwMetricList, cwMetrics...)
+			cwMetricLists = append(cwMetricLists, cwMetrics...)
+			batchCWMetrics(cwMetrics, groupedCWMetricMap, cwMetricsMap)
```

Comment:
Is this the best place to put the batching logic? It seems like we're performing redundant calculations of `createDimensions`. If we have a `groupedMetric` that share the same label set, we should perform dimension processing only once on them together. So, maybe it might be worthwhile taking a look at how we can place the batching logic either deeper within `buildCWMetric` or move the logic for dimension setting out of `getCWMetric` and have it perform on `groupedMetric` after batching.